src/main/java/com/thealgorithms/ciphers/DiffieHellman.java

```diff
@@ -0,0 +1,36 @@
+package com.thealgorithms.ciphers;
+
+import java.math.BigInteger;
+
+public final class DiffieHellman {
+
+    private final BigInteger base;
+    private final BigInteger secret;
+    private final BigInteger prime;
+
+    // Constructor to initialize base, secret, and prime
+    public DiffieHellman(BigInteger base, BigInteger secret, BigInteger prime) {
+        // Check for non-null and positive values
+        if (base == null || secret == null || prime == null || base.signum() <= 0 || secret.signum() <= 0 || prime.signum() <= 0) {
+            throw new IllegalArgumentException("Base, secret, and prime must be non-null and positive values.");
+        }
+        this.base = base;
+        this.secret = secret;
+        this.prime = prime;
+    }
+
+    // Method to calculate public value (g^x mod p)
+    public BigInteger calculatePublicValue() {
+        // Returns g^x mod p
```

Comment:
What is `g`, `x` and `p`?

Reply:
In the Diffie-Hellman key exchange:
1. g represents the base (also called the generator), which is a publicly agreed-upon number used by both parties to perform the key exchange.
2. p is a large prime number, another public value that ensures the security of the key exchange.
3. x is the private secret key chosen by one of the parties. This value is kept secret and used to compute the public key, which is shared with the other party.
Together, these values are used to compute a shared secret without directly transmitting it, ensuring secure communication.